### ISSUE TITLE
Multi-profile YAML documents are not used when put in other than default profile

### ISSUE THREAD

**Issue by michalkarolik** [spring-projects/spring-boot#4132]
We have usecase where we want to use different configurations for some servers, but all of them are 'test' instances ( we have 3 eureka servers and all of them should register to each other). So we are using application-test.yml file with  "Multi-profile YAML documents"

```yaml
---
spring:
  profiles: profile1
  specificProperty: one
---
spring:
  profiles: profile2
  specificProperty: two
```

Having this in place I was sure it will be possible to load run app on specific server with `-Dspring.profiles.active=test,profile1` and on other one with `-Dspring.profiles.active=test,profile2` and correct 'specificProperty' will be used. This is not a case - seems that "Multi-profile YAML documents" works only when put in the main `application.yml` file.

**Comment by shakuzen** [comment]
@philwebb I don't see this issue tagged to a milestone, but it looks like the commit was against master. Should this fix/enhancement also be included in 1.2.7?

**Comment by philwebb** [comment]
I was just waiting for [build 3246](https://build.spring.io/browse/BOOT-PUB-3246) to finish before asking for feedback. I'd rather not change this logic in 1.2.x in case it causes regressions.

@michalkarolik Can you try `1.3.0.BUILD-SNAPSHOT` to see if it fixes your issue?
